### ISSUE TITLE
Control spacing after test expression

### ISSUE THREAD

**Issue by mkahvi** [eslint-stylistic/eslint-stylistic#1205]
### Clear and concise description of the problem

There appears to be no rule controlling spacing after test expressions.

For example the following are valid with no way to control them:
```js
if (foo)console.log("bah");
```
^ no space nor linebreak
```js
if (foo)

  console.log("bah");

else

  console.log("zot")
```
^ excess empty lines

### Suggested solution

Being able to define that I don't want empty lines nor complete lack of spaces would be great, so the examples would be formatted as follows:
```js
if (foo) console.log("bah");
```
```js
if (foo)
  console.log("bah");
else
  console.log("zot")
```

### Alternative

I know of none.

### Additional context

If there's a rule for this, I could not find it.

I tried to use `padding-line-between-statements` for the second case, but it had no effect no matter what I did with it. Notably the following was what I expected to work for it.
```js
'padding-line-between-statements': [
  'error',
  { blankLine: 'never', prev: 'if', next: '*' },
]
```
But that hates `else` so that wouldn't work consistently.

### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue
